### PR TITLE
refactor(pgwire): support complete prepared statement 

### PR DESCRIPTION
**Pull request by ZENOTME**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

support all prepare statement as describe in #5293 
- add the PrepareStatement Abstract 

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Refer to a related PR or issue link (optional)
#5293

### CODE REVIEW

File: src/utils/pgwire/src/pg_extended.rs [hunk 1]
```diff
@@ -356,105 +218,545 @@ impl PgPortal {
     }
 }
 
+/// default_params is to create default params:[String] from type_description:[TypeOid].
+/// The params produced by this function will be used in the PreparedStatement.
+///
+/// type_description is a list of type oids.
```
Comment (BowenXiao1999) [inline src/utils/pgwire/src/pg_extended.rs:224]: We should also mention that the default params is used when user do not provide a value?

File: src/utils/pgwire/src/pg_extended.rs [hunk 2]
```diff
@@ -28,127 +30,23 @@ use crate::pg_response::{PgResponse, StatementType};
...
-                    todo!("can't parse interval type")
-                }
-            };
-            params.push(str)
-        }
-    }
-    Ok(params)
-}
+pub fn replace_params(raw_query: String, params: &[String]) -> String {
+    let mut tmp = raw_query;
```
Comment (BowenXiao1999) [inline src/utils/pgwire/src/pg_extended.rs:46]: Seems like this function is unused?

**Review by BowenXiao1999 — COMMENTED** [review]
LGTM!

Better add more comments on the execution flow. Might be hard to follow for guys do not know the procedure of extended query

File: src/utils/pgwire/src/pg_extended.rs (hunk 1 above)
Comment (ZENOTME) [reply src/utils/pgwire/src/pg_extended.rs:224]: default_params used in instance_default. I add comments in instance_default to illustrate why we need instance_default() and why we use default_params().

Comment (BowenXiao1999) [reply src/utils/pgwire/src/pg_extended.rs:224]: Good!

**Review by BowenXiao1999 — APPROVED** [review]
Most LGTM
